src/cert.rs

```diff
@@ -307,21 +307,18 @@ impl<'a> CrlDistributionPoint<'a> {
             //   a DistributionPoint MUST NOT consist of only the reasons field; either distributionPoint or
             //   cRLIssuer MUST be present.
             match (result.distribution_point, result.crl_issuer) {
-                (None, None) => return Err(Error::MalformedExtensions),
-                _ => {}
+                (None, None) => Err(Error::MalformedExtensions),
+                _ => Ok(result),
             }
-
-            Ok(result)
         })
     }
 
     /// Return the distribution point names (if any).
     #[allow(dead_code)] // TODO(@cpu): remove this once used in CRL validation.
     pub(crate) fn names(&self) -> Result<Option<DistributionPointName<'a>>, Error> {
-        Ok(match self.distribution_point {
-            None => None,
-            Some(der) => Some(DistributionPointName::from_der(der)?),
-        })
+        self.distribution_point
```

Comment:
This way, we can avoid the `None => None` arm.

Reply:
:sweat_smile: I ended up revisiting this in #128 for the same reason:
https://github.com/rustls/webpki/blob/4601fb6cbcc92fa86c029e435009299e43c10123/src/cert.rs#L317-L323